internal/olm/operator/internal/registry_pod.go

```diff
@@ -0,0 +1,189 @@
+		}
+	} else {
+		return tmpPod, nil
+	}
+
+	// Upon creation of new pod, poll and verify that pod status is running
+	podCheck := wait.ConditionFunc(func() (done bool, err error) {
+		p, err := rp.Kubeclient.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+		if err != nil {
+			return true, fmt.Errorf("error getting pod %s: %w", pod.Name, err)
+		}
+		return p.Status.Phase == corev1.PodRunning, nil
+	})
+
+	err = wait.PollImmediateUntil(time.Duration(1*time.Second), podCheck, ctx.Done())
+	if err != nil {
+		return nil, fmt.Errorf("error waiting for registry pod %s to run: %v", pod.Name, err)
+	}
+
+	//TODO: make pod logs available, maybe return log output to caller
+	_, err = rp.fetchPodLogs(ctx, pod)
+	if err != nil {
+		return pod, fmt.Errorf("error in fetching pod logs: %v", err)
+	}
+
+	return pod, err
+}
+
+func (rp *RegistryPod) setDefaults() {
+	// set the grpcPort to default 50051
+	rp.grpcPort = defaultGRPCPort
+
+	if len(rp.IndexImage) == 0 {
+		rp.IndexImage = defaultIndexImage
+	}
+
+	if len(rp.BundleAddMode) == 0 {
+		if rp.IndexImage == defaultIndexImage {
+			rp.BundleAddMode = "semver"
+		} else {
+			rp.BundleAddMode = "replaces"
+		}
+	}
+}
+
+func (rp *RegistryPod) validate() error {
+	if len(rp.BundleImage) == 0 {
+		return fmt.Errorf("bundle image is required")
+	}
+	if len(rp.DBPath) == 0 {
+		return fmt.Errorf("registry database path is required")
+	}
+
+	if len(rp.Namespace) == 0 {
+		return fmt.Errorf("pod namespace is required")
+	}
+
+	return nil
+}
+
+func makePodName(bundleName string) (string, error) {
+	var podName string
+	if strings.Contains(bundleName, "/") {
+		split := strings.Split(bundleName, "/")
+		lastSegment := strings.Split(split[len(split)-1:][0], ":")
+		// get the last but one element from last segment excluding the version number and append index
+		podName = lastSegment[len(lastSegment)-2] + "-index"
+	} else {
+		return "", fmt.Errorf("invalid bundle image name: %s", bundleName)
+	}
+	return podName, nil
+}
+func (rp *RegistryPod) build() (*corev1.Pod, error) {
+	rp.setDefaults()
+	err := rp.validate()
+	if err != nil {
+		return nil, fmt.Errorf("error in validating registry pod: %v", err)
+	}
+
+	podName, err := makePodName(rp.BundleImage)
+	if err != nil {
+		return nil, fmt.Errorf("error in building pod due to invalid bundle image name: %v", err)
+	}
+
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: rp.Namespace,
+			Labels: map[string]string{
+				"app": "default",
+			},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  "registry-grpc",
+					Image: rp.IndexImage,
+					Command: []string{
+						"/bin/sh",
+						"-c",
+						rp.getContainerCmd(),
+					},
+					Ports: []corev1.ContainerPort{
+						{Name: "grpc", ContainerPort: rp.grpcPort},
+					},
+				},
+			},
+		},
+	}
+
+	return pod, nil
+}
+
+// TODO: Use text/template
```

Comment:
Should it be done here or in a follow-up? is it in WIP or not?
If it should be done in a follow-up, could you please clarify the todo action and also use //todo(yournickname): description?

Reply:
Marked as a todo item, for a follow-up PR.

Reply:
Used templating in this PR itself. Removed todo item